### PR TITLE
Implement slider controlpoint visualisation

### PR DESCRIPTION
**Pull request by smoogipoo**
https://streamable.com/zfv9e
https://streamable.com/gxyvz

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Edit/Masks/SliderMasks/Components/ControlPointPiece.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+using osu.Framework.Graphics.Lines;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.Osu.Objects;
+using OpenTK;
+
+namespace osu.Game.Rulesets.Osu.Edit.Masks.SliderMasks.Components
+{
+    public class ControlPointPiece : CompositeDrawable
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Edit/Masks/SliderMasks/Components/ControlPointPiece.cs]: This naming... It needs to change.

**Comment by peppy** [comment]
As a slider gets longer, there seems to be a growing inaccuracy. Is this a known issue?

**Comment by smoogipoo** [comment]
Yeah the inaccurate path is due to the slider's body being redrawn being calculated before the controlpoint calls `Path.Calculate()`.

This'll be fixed later, and requires `SliderPath` to use invalidation.

File: osu.Game.Rulesets.Osu/Edit/Masks/SliderMasks/Components/PathControlPointVisualiser.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+            InternalChild = pieces = new Container<PathControlPointPiece> { RelativeSizeAxes = Axes.Both };
+
+            slider.ControlPointsChanged += _ => updatePathControlPoints();
+            updatePathControlPoints();
+        }
+
+        private void updatePathControlPoints()
+        {
+            while (slider.ControlPoints.Length > pieces.Count)
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Edit/Masks/SliderMasks/Components/PathControlPointVisualiser.cs]: what happens if two points are added here? this seems prone to failure.

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Edit/Masks/SliderMasks/Components/PathControlPointVisualiser.cs]: Explain how that could fail?

Comment (peppy) [reply osu.Game.Rulesets.Osu/Edit/Masks/SliderMasks/Components/PathControlPointVisualiser.cs]: never mind, misread

**Review by peppy — APPROVED** [review]

**Pull request merged**